### PR TITLE
Start rebalancing anomalies - Rock and Flesh anomaly reworked

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR
the primary objective was to refactor the rock anomaly. I rewrote 2 systems to do this, so they are now much more flexible to customize. Meat Anomaly and Stone Anomaly are refactored so that they always have free space near them. The Stone Anomaly is now a geode, in 4 variations for each type of crab and no longer creates crabs during pulses

## Why / Balance
rock anom is hard

## Media
image
image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- tweak: Flesh and Rock anom reworked: It should be easier to maintain them now (I guess)

### CODE REVIEW

**Comment by brainfood1183** [comment]
Interested to see how rock anom turns out, might be a little on easy side now but we will see. 

Not sure how this will effect the meat anomaly.

File: Content.Server/Anomaly/Effects/EntityAnomalySystem.cs [hunk 1]
```diff
@@ -22,55 +22,69 @@ public override void Initialize()
...
+        if (entry.Spawns.Count == 0)
             return;
 
-        if (!_map.TryGetGrid(xform.GridUid, out var grid))
+        var xform = Transform(component.Owner);
+        if (!TryComp<MapGridComponent>(xform.GridUid, out var grid))
             return;
 
+        var amount = (int) (MathHelper.Lerp(entry.MinAmount, entry.MaxAmount, 1) + 0.5f);
+
```
Comment (metalgearsloth) [inline Content.Server/Anomaly/Effects/EntityAnomalySystem.cs:84]: Is this supposed to be random.

File: Content.Server/Anomaly/Effects/TileAnomalySystem.cs [hunk 2]
```diff
@@ -17,25 +19,92 @@ public sealed class TileAnomalySystem : EntitySystem
...
+
         var localpos = xform.Coordinates.Position;
         var tilerefs = grid.GetLocalTilesIntersecting(
-            new Box2(localpos + new Vector2(-radius, -radius), localpos + new Vector2(radius, radius)));
+            new Box2(localpos + new Vector2(-entry.MaxRange, -entry.MaxRange), localpos + new Vector2(entry.MaxRange, entry.MaxRange))).ToArray();
+
+        if (tilerefs.Length == 0)
+            return;
+
+        _random.Shuffle(tilerefs);
```
Comment (metalgearsloth) [inline Content.Server/Anomaly/Effects/TileAnomalySystem.cs:89]: Don't copypaste between systems make it reusable.

File: Content.Shared/Anomaly/Effects/Components/EntitySpawnAnomalyComponent.cs [hunk 3]
```diff
@@ -4,50 +4,69 @@ namespace Content.Shared.Anomaly.Effects.Components;
...
-    [DataField, ViewVariables(VVAccess.ReadWrite)]
-    public bool SpawnOnStabilityChanged = false;
+    [DataField]
+    public bool SpawnOnStabilityChanged { get; init; } = false;
+
+    /// <summary>
+    /// Whether or not anomaly spawns entities on SeverityChanged
+    /// </summary>
+    [DataField]
+    public bool SpawnOnSeverityChanged { get; init; } = false;
```
Comment (metalgearsloth) [inline Content.Shared/Anomaly/Effects/Components/EntitySpawnAnomalyComponent.cs:71]: Use access attribute not the init property.

File: Content.Server/Anomaly/Effects/EntityAnomalySystem.cs (hunk 1 above)
Comment (TheShuEd) [reply Content.Server/Anomaly/Effects/EntityAnomalySystem.cs:84]: oops

File: Content.Server/Anomaly/Effects/TileAnomalySystem.cs (hunk 2 above)
Comment (TheShuEd) [reply Content.Server/Anomaly/Effects/TileAnomalySystem.cs]: I don't know the best way to do this. It was copying a lot of code before refactor, so maybe it should be like this for now.

**Pull request merged**